py-version-control.Rmd

```diff
@@ -0,0 +1,428 @@
+and can't be automatically reconciled. 
+
+We can create a conflict like this by making changes in our `README.md` file. 
+First, make sure that your local repository and Github repository are synced 
+up - that you have pushed all your local changes to Github and pulled all your 
+Github changes to your local computer. 
+
+Then, **on your computer**, make these changes to the `README.md` file: 
+
+```
+## Meeting time
+
+- December 21, 2019
+- 18:00 UTC
+```
+
+Commit the changes as we've done before but do **not** push them to Github. 
+
+Now, go to your repository on Github, click on the `README.md` file and edit 
+it as follows: 
+
+```
+## Meeting time
+
+- November 4, 2018
+- 18:00 UTC
+```
+
+Commit the changes. 
+
+Now we have conflicting changes in our repository -- one in the Github copy 
+and one on our local computer copy. 
+
+We want to sync our changes by either pushing from the copy on our computer to 
+the Github copy or by pulling from the Github copy to our computer. It turns 
+out that we can't do the first option (try it and see what happens), so we'll 
+have to use the second option, of pulling from Github to our computer. When 
+we try this, we should get a few red boxes that pop up saying "Pull failed"
+and/or "Merge failed". 
+
+> Note that, when we try to push, we *do* have the option of using a "force 
+push" to override whatever changes we made in Github. If you're absolutely 
+sure that there's nothing in the Github copy you want, you can try this, but 
+it's generally a good idea to pull down the changes and look at the conflict, 
+just to make sure. 
+
+Because of the conflicting changes we made, the update of our 
+local repository failed. Git doesn't know 
+which changes we really want and has pushed us into a temporary "conflict" 
+state until we tell it specificallly which changes we want. 
+
+To do this, click on "View conflict" (should be on the top right of the 
+GitKraken window). Click on the file in conflict (`README.md`). This will 
+open up a different view, where the two conflicting versions of the file 
+are shown side by side. You can check the box picking one version, the other, 
+or both, with the combined output shown in the window below. Once you've 
+picked what you're happy with, click the "Save" button on the top right, close 
+the file by clicking the "X" and finish the update by clicking on 
+"Commit and merge."
+
+TODO: screenshots of this process
+
+## Using Local vs Remote Repositories {#py-version-control-local-vs-remote}
+
+In the previous section, we set up an online copy of our project repository. 
+This wasn't too risky because our repository didn't have any large file or 
+confidential information. However, for some projects, that will be the case, 
+which is why it's important to think carefully before creating an online 
+repository. 
+
+Consider the following before you put it in an online repository: 
+- Size
```

Comment:
This bullet list also doesn't format correctly when I use `make html` to view the lesson.